### PR TITLE
PIPE2D/PIPE-813 Make str more useful for pfsConfig enums

### PR DESCRIPTION
**Pull request by RobertLuptonTheGood**

### CODE REVIEW

File: python/pfs/datamodel/pfsConfig.py [hunk 1]
```diff
@@ -34,6 +35,15 @@ def getFitsHeaders(cls):
         keyBase = "HIERARCH " + cls.__name__ + "."
         return {keyBase + member.name: (member.value, member.__doc__) for member in cls}
 
+    def __str__(self):
+        """Return the enum's name"""
+        try:
+            name, val = re.search(r"^<[^.]+.([^:]+): +(\d+)>$", repr(TargetType.SKY)).groups()
```
Comment (PaulPrice) [inline python/pfs/datamodel/pfsConfig.py:41]: Why is `TargetType.SKY` hard-wired?

Comment (hassanxp) [inline python/pfs/datamodel/pfsConfig.py:41]: May be optimal to compile regex in advance, if ``str()`` is going to be used multiple times.

Comment (RobertLuptonTheGood) [reply python/pfs/datamodel/pfsConfig.py:41]: That's a micro optimisation that I have never seen show up in a profiler.  It'd add state to the class, always bad, for almost no gain.

Comment (RobertLuptonTheGood) [reply python/pfs/datamodel/pfsConfig.py:41]: Cut and paste stupidity.

**Review by PaulPrice — APPROVED** [review]

**Pull request merged**